dbt_subprojects/dex/models/trades/sonic/platforms/metropolis_sonic_base_trades.sql

```diff
@@ -0,0 +1,23 @@
+
+{{ config(
+    schema = 'metropolis_sonic',
+    alias  = 'base_trades',
+    materialized = 'incremental',
+    file_format  = 'delta',
+    incremental_strategy = 'merge',
+    unique_key = ['tx_hash', 'evt_index'],
+    incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+) }}
+
+
+   {{
+    metropolis_compatible_trades(
```

Comment:
there is no indication that this metropolis logic will be used heavily by other DEXs/models in the future. the macro includes a harcoded CTE which is specific to metropolis on sonic. let's remove the need for a macro and put that SQL logic in this model file directly.

Reply:
yes, move code from macro to model, delete macro.
based on your error there, it looks like you did not remove reference to macro in the model so it's failing while looking for it, which was deleted. there should be no macro call.